src/pbt/deployment/jobs/databricks.py

```diff
@@ -901,46 +901,58 @@ def deploy(self):
                 count = sum(len(v) for v in self.pipeline_configurations.values())
                 log(f"\n\n{Colors.OKBLUE} Uploading {count} pipeline configurations {Colors.ENDC}\n\n")
 
-            def execute_job(_fabric_id, config_content, config_path):
+            def execute_job(_fabric_id, pipeline_id, config_name, config_content):
                 futures.append(
                     executor.submit(
                         lambda f_id=str(
                             _fabric_id
-                        ), conf_content=config_content, conf_path=config_path: self._upload_configuration(
-                            f_id, conf_content, conf_path
+                        ), p_id=pipeline_id, conf_name=config_name, conf_content=config_content: self._upload_configuration(
+                            f_id, p_id, conf_name, conf_content
                         )
                     )
                 )
 
             for pipeline_id, configurations in self.pipeline_configurations.items():
-                path = self.project_config.system_config.get_dbfs_base_path()
-                pipeline_path = (
-                    f"{path}/{self.project.project_id}/{self.project.release_version}/configurations/{pipeline_id}"
-                )
 
                 for configuration_name, configuration_content in configurations.items():
-                    configuration_path = f"{pipeline_path}/{configuration_name}.json"
-
                     for fabric_id in self.project_config.fabric_config.db_fabrics():
-                        execute_job(fabric_id, configuration_content, configuration_path)
+                        execute_job(fabric_id, pipeline_id, configuration_name, configuration_content)
 
         return await_futures_and_update_states(futures, self._STEP_ID)
 
-    def _upload_configuration(self, fabric_id, configuration_content, configuration_path):
+    def _upload_configuration(self, fabric_id, pipeline_id, config_name, configuration_content):
+        def base_path(fab_id: Optional[str]):
+            path = self.project_config.get_db_base_path(fab_id)
+            pipeline_path = (
+                f"{path}/{self.project.project_id}/{self.project.release_version}/configurations/{pipeline_id}"
+            )
+            return pipeline_path
+
         try:
+            # we are creating path and then uploading the content
+            # we need to upload for the scenerios
+            # in case volumes is set or not.
+            # after some versions most likely databricks will deprecate the dbfs one.
             client = self.databricks_jobs.get_databricks_client(str(fabric_id))
+            base_p = base_path(None)  # old dbfs path first.
+            configuration_path = f"{base_p}/{config_name}.json"
```

Comment:
In case user supplies a non-volume dbfs path in the fabric, we will be unnecessarily uploading to two non-volume dbfs paths. Can we avoid that?